Use spf13/cobra for docker network and subcommands

vdemeester opened the pull request:
- Migrates `network` command and subcommands (`connect`, `create`, `disconnect`, `inspect`, `list` and `remove`) to spf13/cobra 🐍.
- Create a RequiredExactArgs helper function for command that require an exact number of arguments.

/cc @dnephin @thaJeztah @LK4D4 

🐸

Signed-off-by: Vincent Demeester vincent@sbr.pm

dnephin commented on api/client/network/connect.go:
```diff
@@ -0,0 +1,65 @@
+		if _, err := runconfigopts.ValidateLink(link); err != nil {
+			return err
+		}
+	}
```
I think it would be better to continue to use `opts.NewListOpts(runconfigopts.ValidateLink)` for this.  It's nice when all the flag/arg validation happens at the same spot, so the error messages are consistent (you get the usage message). You can use it with `flags.Var(&opts.links, ...)`.

With this setup I don't think you'll get the usage info.

vdemeester replied:
ah yes :+1:

dnephin commented on cli/required.go:
```diff
@@ -40,3 +40,19 @@ func RequiresMinArgs(min int) cobra.PositionalArgs {
 }
+
+// RequiresExactArgs returns an error if there is not the exact number of args
+func RequiresExactArgs(number int) cobra.PositionalArgs {
```
Nice! I was thinking maybe `ExactArgs(number int)`, what do you think?

vdemeester replied:
Sounds better, I'll update :wink:

thaJeztah commented:
built and tried this, and all looks to be working correctly 👍

thaJeztah commented:
ping @dnephin PTAL, its green 💚

dnephin commented:
LGTM!

vdemeester commented:
Rebased :angel:

vdemeester committed:
Migrate network command to cobra

- Migrates network command and subcommands (connect, create, disconnect,
  inspect, list and remove) to spf13/cobra
- Create a RequiredExactArgs helper function for command that require an
  exact number of arguments.

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

thaJeztah commented:
LGTM

Janky is green, but didn't notify GitHub; https://jenkins.dockerproject.org/job/Docker-PRs/28337/console

```
docker-pr28337
09:37:46 Archiving artifacts
09:37:47 Notifying endpoint 'HTTP:https://leeroy.dockerproject.org/notification/jenkins'
09:37:49 [WS-CLEANUP] Deleting project workspace...[WS-CLEANUP] done
09:37:49 Finished: SUCCESS
```
